extras/create_snapshots_reports_scorecard.py

```diff
@@ -365,127 +376,135 @@ def generate_weekly_snapshots(db, cyhy_db_section):
+                    "{}/cyhy-reports:stable".format(NCATS_DHUB_URL),
+                    "cyhy-report",
+                    "--nolog",
+                    "--cyhy-section",
+                    cyhy_db_section,
+                    "--scan-section",
+                    scan_db_section,
+                    "--final",
+                    "--encrypt",
+                    org_id,
+                ],
+                stdout=subprocess.PIPE,
+                stdin=subprocess.PIPE,
+                stderr=subprocess.PIPE,
             )
-            successful_reports.append(i)
         else:
-            logging.info(
-                "[%s] Failure to generate report: %s",
-                threading.current_thread().name,
-                i,
+            p = subprocess.Popen(
+                [
+                    "docker",
+                    "run",
+                    "--rm",
+                    "--volume",
+                    "/etc/cyhy:/etc/cyhy",
+                    "--volume",
+                    "{}:/home/cyhy".format(CYHY_REPORT_DIR),
+                    "{}/cyhy-reports:stable".format(NCATS_DHUB_URL),
+                    "cyhy-report",
+                    "--cyhy-section",
+                    cyhy_db_section,
+                    "--scan-section",
+                    scan_db_section,
+                    "--final",
+                    "--encrypt",
+                    org_id,
+                ],
+                stdout=subprocess.PIPE,
+                stdin=subprocess.PIPE,
+                stderr=subprocess.PIPE,
             )
-            logging.info(
-                "[%s] Stderr report detail: %s%s",
-                threading.current_thread().name,
-                data,
-                err,
+    else:
+        if nolog:
+            p = subprocess.Popen(
+                [
+                    "cyhy-report",
+                    "--nolog",
+                    "--cyhy-section",
+                    cyhy_db_section,
+                    "--scan-section",
+                    scan_db_section,
+                    "--final",
+                    "--encrypt",
+                    org_id,
+                ],
+                stdout=subprocess.PIPE,
+                stdin=subprocess.PIPE,
+                stderr=subprocess.PIPE,
             )
-            failed_reports.append(i)
+        else:
+            p = subprocess.Popen(
+                [
+                    "cyhy-report",
+                    "--cyhy-section",
+                    cyhy_db_section,
+                    "--scan-section",
+                    scan_db_section,
+                    "--final",
+                    "--encrypt",
+                    org_id,
+                ],
+                stdout=subprocess.PIPE,
+                stdin=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+    data, err = p.communicate()
+    report_time = time.time() - report_time
+    report_durations.append((org_id, report_time))
```

Comment:
Again but this time about `rd_lock`.

Reply:
Excellent catch - please see https://github.com/cisagov/cyhy-reports/pull/106/commits/c13a86bb3965934b98f7e5f114c6d1c0616899d5.